### PR TITLE
Update libsys-webforms to rails 5 and ruby-2.6.3

### PR DESCRIPTION
**Pull request by jgreben**
- Updates gems, and gemfile
- Updates rubocop and fixes failing tests and updates tests for stricter rspec cops
- Updates travis config to speed up build
- Updates oracle initializer to monkey patch database types for legacy back-end database

### CODE REVIEW

**Comment by jgreben** [comment]
Currently can not deploy to dev. See https://github.com/sul-dlss/operations-tasks/issues/2023

File: app/controllers/packages_controller.rb [hunk 1]
```diff
@@ -41,7 +41,7 @@ def create
     update_special_attributes
     # vendor_id_write isn't actually used by vnd_load_data_eload but the value
     # 001 is written to the table
-    package_params.store(:vendor_id_write, '001')
+    package_params[:vendor_id_write] = '001'
```
Comment (shelleydoljack) [inline app/controllers/packages_controller.rb:44]: Looks good. I wonder why we were using the store method. For package creation it doesn't seem to matter to have used store versus simply parameter assignment.

**Comment by jgreben** [comment]
To get this working on the server:
- add `profile:devtools` to puppet config and update to ruby-2.6.3 (PR of https://github.com/sul-dlss/puppet/tree/jgrebenAddsDevToolSetProfile)
- `cap dev ssh`, then 
  - `ksu`
  - `rvm get stable && rvm reload && rvmsudo rvm repair all`
  - check `/etc/httpd/conf.d/zpassenger.load` is using ruby-2.6.3
  - check `/etc/httpd/conf.d/passenger_extra.conf` is using ruby-2.6.3
  - may need to run ` /usr/local/rvm/bin/rvm ruby-2.6.3 exec passenger-install-apache2-module -a` if puppet does not update these.

File: spec/models/edi_lin_spec.rb [hunk 2]
```diff
@@ -1,6 +1,8 @@
 require 'rails_helper'
 
 RSpec.describe EdiLin, type: :model do
+  # The order of the tests matters here
```
Comment (shelleydoljack) [inline spec/models/edi_lin_spec.rb:4]: Nice to have a comment here about disabling the cop. Thanks!

File: config/initializers/oracle.rb [hunk 3]
```diff
@@ -1,4 +1,8 @@
 if Rails.env.development? || Rails.env.production?
+  if Rails.env.development?
```
Comment (shelleydoljack) [inline config/initializers/oracle.rb:2]: Why is it better to do it here?

File: config/initializers/oracle.rb [hunk 4]
```diff
@@ -24,6 +24,63 @@
 
       # Initialize ruby-plsql gem here
       plsql.activerecord_class = ActiveRecord::Base
+
+      # See https://github.com/rsim/oracle-enhanced/issues/1679
```
Comment (shelleydoljack) [inline config/initializers/oracle.rb:28]: 👍

File: app/controllers/packages_controller.rb (hunk 1 above)
Comment (jgreben) [reply app/controllers/packages_controller.rb:44]: The Googles probably told us to do it that way, and then rubocop told us not to!

File: config/initializers/oracle.rb (hunk 3 above)
Comment (jgreben) [reply config/initializers/oracle.rb:2]: Before it was in the Gemfile, and rubocop complains about having stuff in there twice (in different groups). Also, before we had to add the `activerecord-oracle_enhanced-adapter` into the development group in order to run some rake commands. This way we no longer have to do that.

File: app/controllers/packages_controller.rb (hunk 1 above)
Comment (shelleydoljack) [reply app/controllers/packages_controller.rb:44]: Possibly was an issue in rails 4.2.7 to assign a value to a parameter that wasn't passed from the view to the controller. Assignment seems to work in rails 5.

File: app/controllers/concerns/symphony_cgi.rb [hunk 5]
```diff
@@ -23,7 +24,8 @@ def request_url(script, cgi_params)
   end
 
   def query(cgi_params)
-    URI.encode_www_form(cgi_params).split('&').select { |param| param =~ /=./ }.join('&')
+    cgi_params.delete_if { |k, v| v.nil? }
```
Comment (jgreben) [inline app/controllers/concerns/symphony_cgi.rb]: Need to look at what is going on here per https://github.com/sul-dlss/libsys-webforms/issues/612

File: app/controllers/shelf_sel_searches_controller.rb [hunk 6]
```diff
@@ -3,7 +3,7 @@
 ###
 class ShelfSelSearchesController < ApplicationController
   def delete_saved_search
-    ShelfSelSearch.find([params[:user], params[:search_name]]).delete
-    render nothing: true
+    ShelfSelSearch.find([params[:user_name], params[:search_name]]).delete
```
Comment (jgreben) [inline app/controllers/shelf_sel_searches_controller.rb:6]: Why did this param change, and does deleting a saved search form still work after this change?

**Review by jgreben — COMMENTED** [review]
Let's check into and fix some things.

File: app/controllers/shelf_sel_searches_controller.rb (hunk 6 above)
Comment (jgreben) [reply app/controllers/shelf_sel_searches_controller.rb:6]: Updated for consistency, and to fix broken delete functionality:  https://github.com/sul-dlss/libsys-webforms/blob/ce60e24076a8309629cc62888cf7c310029eb48d/app/assets/javascripts/shelf_selection_reports.js#L56

File: app/controllers/concerns/symphony_cgi.rb (hunk 5 above)
Comment (jgreben) [reply app/controllers/concerns/symphony_cgi.rb]: Fixed https://github.com/sul-dlss/libsys-webforms/issues/612

File: app/controllers/packages_controller.rb [hunk 7]
```diff
@@ -96,7 +96,7 @@ def destroy
   def run_tests() end
 
   def list_transfer_logs
-    @logs = VndRunlog.all.where('run_date > ?', Time.now.getlocal - 1)
+    @logs = VndRunlog.all.where('run_date > ?', Time.now.getlocal.strftime('%Y-%m-%d'))
```
Comment (jgreben) [inline app/controllers/packages_controller.rb]: See https://github.com/sul-dlss/libsys-webforms/issues/613

**Comment by tallenaz** [comment]
If possible, I'd like us to  hook -dev and -prod up to [rails-latest](https://github.com/sul-dlss/puppet/blob/production/hieradata/techstack/rails-latest.eyaml) or rails-stabe, which lets us upgrade more than one rails machine's components in one place, and would get us up to the latest versions of ruby, rvm, and passenger.

**Comment by dlrueda** [comment]
Doing anything that will automatically update prod without first thoroughly testing on -dev makes me nervous, mostly because of the Oracle db issue.  I feel that we have a non-standard rails implementation.  Not sure if what you're suggesting above is anything automatic.

**Comment by tallenaz** [comment]
Looking a little more closely, the pattern that access uses is: put -dev and -stage on rails-latest and put -prod on rails-stable. I'll confirm, but without talking to access folks, my sense is that they make changes to rails-latest, test, and then roll what passes out to rails-stable, to avoid just that issue. If we did this, we'd want to make sure all parties know when what gets changed.

**Review by shelleydoljack — APPROVED** [review]

**Pull request merged**